pkg/segment/reader/segread/agiletreereader.go

```diff
@@ -155,9 +155,16 @@ func (str *AgileTreeReader) ReadTreeMeta() error {
 		return err
 	}
 
-	if str.metaFileBuffer[0] != utils.STAR_TREE_BLOCK[0] {
-		log.Errorf("AgileTreeReader.ReadTreeMeta: received an unknown encoding type for agileTree: %v",
-			str.metaFileBuffer[0])
+	encodingVersion := str.metaFileBuffer[0]
+
+	switch encodingVersion {
+	case utils.STAR_TREE_BLOCK[0]:
+		// do nothing, this is the expected encoding
+	case utils.STAR_TREE_BLOCK_LEGACY[0]:
+		log.Errorf("AgileTreeReader.ReadTreeMeta: received a legacy encoding type for agileTree: %v", encodingVersion)
+		return errors.New("received legacy agileTree encoding")
+	default:
+		log.Errorf("AgileTreeReader.ReadTreeMeta: received an unknown encoding type for agileTree: %v", encodingVersion)
 		return errors.New("received non-agileTree encoding")
 	}
```

Comment:
nit: moving this to a helper function could clean up the code a little since this function is fairly long. But you can keep this if you want